js/interface.js

```diff
@@ -128,14 +132,25 @@ function getDataSources() {
 
 function renderDataSources(dataSources) {
   var html = [];
-
   dataSources.forEach(function (dataSource) {
     html.push(getDataSourceRender(dataSource));
   });
 
   $dataSources.html(html.join(''));
   $initialSpinnerLoading.removeClass('animated');
   $contents.removeClass('hidden');
+  $('#trash-sources').hide();
+}
+
+function renderTrashSources(trashSources) {
+  var html = [];
```

Comment:
@upplabs-alex-levchenko Please check with @AndrRyaz on all the usual rules of extra such as empty lines as well as all the linting rules.

```suggestion
  var html = [];

```

Reply:
@upplabs-alex-levchenko Please confirm that you have instructed @AndrRyaz about the rules in the use of empty lines and repeat the rules here in a comment so I can confirm it's correct, because I still see many more instances in this PR that violates the rules that other developers seem to have learnt quite successfully.

Reply:
@upplabs-alex-levchenko Please read and respond to the comment above. There are still plenty of errors that other developers seem to have learnt not to make, e.g. mandatory empty lines etc.